### PR TITLE
🏁 Starts have types part 1

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/flows/views.py [hunk 1]
```diff
@@ -2037,7 +2037,7 @@ def __init__(self, *args, **kwargs):
                 help_text=_("Include contacts currently active in a flow"),
             )
 
-            start_type = forms.ChoiceField(
+            recipients_mode = forms.ChoiceField(
```
Comment (rowanseymour) [inline temba/flows/views.py:2040]: having this also be called `start_type` seemed a bit confusing

File: templates/flows/flow_broadcast.haml [hunk 2]
```diff
@@ -139,7 +139,7 @@
     var queryField = document.querySelector('.query');
     var recipientsField = document.querySelector('.omnibox');
 
-    var startTypeSelect = document.querySelector("temba-select[name='start_type']");
+    var startTypeSelect = document.querySelector("temba-select[name='recipients_mode']");
```
Comment (norkans7) [inline templates/flows/flow_broadcast.haml:142]: `recipientModeSelect` variable name

File: temba/flows/models.py [hunk 3]
```diff
@@ -3116,6 +3117,12 @@ class FlowStart(models.Model):
         (STATUS_FAILED, _("Failed")),
     )
 
+    TYPE_MANUAL = "M"
+    TYPE_API = "A"
+    TYPE_FLOW_ACTION = "F"
+
+    TYPE_CHOICES = ((TYPE_MANUAL, "Manual"), (TYPE_API, "API"), (TYPE_FLOW_ACTION, "Flow Action"))
```
Comment (rowanseymour) [inline temba/flows/models.py:3124]: Looking at mailroom.. starts are also used internally to handle IVR and scheduled triggers..  could add `TYPE_TRIGGER`..  or `TYPE_IVR_TRIGGER` and `TYPE_SCHEDULE_TRIGGER`

@nicpottier ?

Comment (nicpottier) [reply temba/flows/models.py:3124]: sure

**Pull request merged**